### PR TITLE
Group manager

### PR DESCRIPTION
**Pull request by KennethRuan**

### CODE REVIEW

File: groups/api.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from pathlib import Path
+from groups.services import get_group, create_group, update_group, delete_group
+
+from typing import List
+from .models import Group, GroupSkeleton
+
+router = Router()
+
+@router.get("/", response=List[Group])
+def get_group_handler(request, group_id: int = None, member_id: int = None):
```
Comment (emmahuangg) [inline groups/api.py]: This is so smart, I'll name my routes _handler from now on too!

File: groups/services.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from multiset.db_utils import execute_query
+from pathlib import Path
+from typing import List
+
+from .models import Group, GroupSkeleton
+
+def create_group(group: GroupSkeleton, user_ids: List[int]):
+    # user_ids_clause = ",".join([f"{user_id}" for user_id in user_ids])
```
Comment (emmahuangg) [inline groups/services.py]: Remove these comments?

File: groups/services.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    created_group = execute_query(
+        Path("groups/sql/create_group.sql"),
+        {"name": group.name, "optimize_payments": group.optimize_payments, "budget": group.budget, "user_ids": user_ids},
+    )
+    # print("created_group", created_group, group.name, group.optimize_payments, user_ids)
```
Comment (emmahuangg) [inline groups/services.py]: Remove

**Review by emmahuangg — CHANGES_REQUESTED** [review]
I sent some errors i'm getting in our backend channel :)

File: groups/api.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+def update_group_handler(request, group: Group):
+    """
+    Updates a group in the database.
+    Args:
+        group: the group to be updated
+    Returns:
+        a JSON response with the status of the operation
+    """
+    try:
+        ret = update_group(group)
```
Comment (emmahuangg) [inline groups/api.py]: do we need to return ret here? i think update_group returns null if the update is successful, which isn't very helpful to return to frontend

**Review by emmahuangg — CHANGES_REQUESTED** [review]
DELETE isn't working for me, not sure if it's just on my end?

File: groups/api.py (hunk 4 above)
Comment (KennethRuan) [reply groups/api.py]: You're right, I'll remove it!

**Review by benymng — APPROVED** [review]
LGTM

**Review by benymng — CHANGES_REQUESTED** [review]
When I put a user ID of 1, on the getGroups endpoint the group only has that user in the group. I think the intended behaviour is that the group has everyone including that user

**Review by emmahuangg — APPROVED** [review]
LGTM!

**Review by benymng — APPROVED** [review]

**Pull request merged**